gems/aws-sigv4/spec/signer_spec.rb

```diff
@@ -270,6 +271,29 @@ module Sigv4
 
       end
 
+      context '#sign_event' do
```

Comment:
Are there any partial event receive cases we need to handle here, or is that abstracted out before we get to this phase? Defensive coding here is a good idea.

Reply:
Yeah, currently no. This event signature v4 signing is currently used for event sending only and received events only requires eventstream decoding.